yolo_to_yolo/scripts.py

```diff
@@ -0,0 +1,61 @@
+import os
+import shutil
+import re
+from tkinter import Tk, filedialog
+
+# Function to open a folder dialog
+def select_directory(title):
+    root = Tk()
+    root.withdraw()  # Hide the main window
+    directory = filedialog.askdirectory(title=title)
+    root.destroy()
+    return directory
+
+# Function to copy files
+def copy_files(input_dir, output_dir, end_number, even_only=False):
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+    
+    # Adjusted regex pattern to match the number at the end after the last underscore
+    pattern = re.compile(r'.*_([0-9]+)\.')
+    
+    for filename in os.listdir(input_dir):
+        match = pattern.match(filename)
+        if match:
+            file_number = int(match.group(1))
+            if file_number <= end_number and (not even_only or file_number % 2 == 0):
+                src = os.path.join(input_dir, filename)
+                dst = os.path.join(output_dir, filename)
+                shutil.copy2(src, dst)
+
+# Main function
+def main():
```

Comment:
From what I can see, this would only copy the images, right? We need to copy the labels over as well. That's why I said you should use the reader and writer. They would take care of the associations and formatting for you, in addition to familiarizing you with the codebase.

Reply:
Also, this functionality should be implemented as a `YoloDataTransformer`. You can just maintain an index and return when you've passed enough through without having to exhaust the entire input `Iterable`. That class should only be like 5 lines code. Then, you can just do this for the script.

```python
def copy_subset(source: Path, destination: Path, count: int, skip_odd: bool):
    pipeline = YoloDataPipeline(
        [SelectFirstN(count, skip_odd)]
    )
    
    pipeline.apply_to_dir(source, destination)
```